blog/_posts/2024-08-02-istio-with-gateway-api.md

```diff
@@ -0,0 +1,230 @@
+---
+title: "Using AKS-managed Istio External Ingress Gateway with Gateway API"
+description: "Learn how to use the Gateway API to manage the Istio external ingress gateway on AKS."
+date: 2024-08-06
+author: Paul Yu
+categories: networking, operations, add-ons
+---
+
+## Introduction
+
+Kubernetes is great at orchestrating containers, but it can be a bit tricky to manage traffic routing. There are many options and implementations that you, as a cluster operator has probably had to deal with. We have the default Services resource that can be used to expose applications, but it is limited to routing based on layer 4 (TCP/UDP) and does not support more advanced traffic routing use cases. There's also the Ingress controller, which enabled layer 7 (HTTP) routing, and securing the North-South traffic using TLS, but it was not standardized and each vendor implementation learning a new set of resource annotations. When it comes to managing and securing East-West traffic between services, there's Service Mesh which is yet another layer of infrastructure to manage on top of Kubernetes. And we're in the same boat when it comes to resource management with each vendor having their own ways of doing things.
+
+## Rise of the Gateway API
+
+There's no doubt that Ingress and Service Mesh have greatly improved the way we manage and secure traffic in Kubernetes, but the subtle differences between implementations has made it difficult to standardize on a single way of doing things. This is where the [Gateway API](https://gateway-api.sigs.k8s.io/) comes in. It is a Kubernetes add-on that aims to bring users an extensible, role-oriented, and protocol-aware services configuration mechanism. It is a new API that both Ingress and Service Mesh vendors can standardize on.
+
+The API has been designed from the ground up, taking into account the lessons learned from the Ingress and Service Mesh APIs. To me, it is a game-changer because ensures that you can implement express traffic routing configurations that were only possible via custom Ingress annotations. It also offers role-oriented configuration, which means that cluster operators can configure gateways and app developers can manage routes through the gateway. This is a big win for organizations that have a separation of duties between cluster operators and app developers. And finally, the API is portable, meaning that the API specification is supported by many implementations including [Istio](https://istio.io/latest/).
```

Comment:
```suggestion
The API has been designed from the ground up, taking into account the lessons learned from the Ingress and service mesh APIs. To me, it is a game-changer because ensures that you can express traffic routing configurations that were previously only possible via custom Ingress annotations. It also offers role-oriented configuration, which means that cluster operators can configure gateways and app developers can manage routes through the gateway. This is a big win for organizations that have a separation of duties between cluster operators and app developers. And finally, the API is portable, meaning that the API specification is supported by many implementations including [Istio](https://istio.io/latest/).
```